chain/actors/builtin/market/v0.go

```diff
@@ -0,0 +1,275 @@
+package market
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/chain/actors/adt"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	v0adt "github.com/filecoin-project/specs-actors/actors/util/adt"
+	typegen "github.com/whyrusleeping/cbor-gen"
+)
+
+type v0State struct {
+	market.State
+	store adt.Store
+}
+
+func (s *v0State) TotalLocked() (abi.TokenAmount, error) {
+	fml := types.BigAdd(s.TotalClientLockedCollateral, s.TotalProviderLockedCollateral)
+	fml = types.BigAdd(fml, s.TotalClientStorageFee)
+	return fml, nil
+}
+
+func (s *v0State) BalancesChanged(otherState State) bool {
+	v0otherState, ok := otherState.(*v0State)
+	if !ok {
+		// there's no way to compare differnt versions of the state, so let's
+		// just say that means the state of balances has changed
+		return true
+	}
+	return !s.State.EscrowTable.Equals(v0otherState.State.EscrowTable) || !s.State.LockedTable.Equals(v0otherState.State.LockedTable)
+}
+
+func (s *v0State) StatesChanged(otherState State) bool {
+	v0otherState, ok := otherState.(*v0State)
+	if !ok {
+		// there's no way to compare differnt versions of the state, so let's
+		// just say that means the state of balances has changed
```

Comment:
This isn't _entirely_ true, but it may not be worth doing this correctly. We can, in fact, do this the slow way and compare everything.